cpp/tests/core/Tensor.cpp

```diff
@@ -2801,5 +2801,70 @@ TEST_P(TensorPermuteDevices, NumpyIO) {
     utility::filesystem::RemoveFile(file_name);
 }
 
+TEST_P(TensorPermuteDevices, RValueScalar) {
+    const core::Device &device = GetParam();
+    core::Tensor t, t1;
+
+    // Assign scalars basic tests.
+    t = core::Tensor::Init<uint8_t>({{1, 2, 3, 4}, {5, 6, 7, 8}}, device);
+    t1 = core::Tensor::Init<uint8_t>({{1, 10, 3, 4}, {5, 6, 7, 8}}, device);
+    t[0][1] = 10;
```

Comment:
If we look back to the `operator=` function that we modified, the purpose of this PR is to enable "assigning scalar value to a tensor of arbitrary shape", where the scalar value will broadcast to all elements in the tensor.  Previously it is only possible to assign scalar to a tensor with one element.

Therefore, we should test "assigning scalar value to a tensor of arbitrary shape", not assigning to a sliced tenor with only one element (e.g. `t[0][1]` has only one element, this is already supported before this PR).

I suggest re-writing all tests and do it more systematically
- First, we should test different shapes, e.g. create tensors with different shapes, e.g. `{}`, `{0}`, `{0, 1}`, `{1}`, `{1, 1}` `{1, 2}` and assign to the full tensor. 
- We shall also test dtype conversion, e.g. what if you assign a float value to an int tensor? Will it throw exception, or will it do the conversion automatically?

Reply:
Sure. Thanks for pointing this out. I've modified the tests with these changes and added few more.